packages/scandipwa/src/component/ProductCard/ProductCard.component.js

```diff
@@ -243,23 +243,45 @@ export class ProductCard extends Product {
     requiresConfiguration() {
         const {
             product: {
-                type_id: type,
                 options = [],
+                items = [],
                 links_purchased_separately
             }
         } = this.props;
 
-        const configureBundleAndGrouped = type === PRODUCT_TYPE.bundle || type === PRODUCT_TYPE.grouped;
-        const configureConfig = (type === PRODUCT_TYPE.configurable
-            && Object.keys(super.getConfigurableAttributes())
-                .length !== Object.keys(this.getConfigurableAttributes()).length)
-            || (type === PRODUCT_TYPE.configurable
-               && Object.values(this.getConfigurableAttributes()).some((value) => value.attribute_values.length === 0));
-        const configureCustomize = options.some(({ required = false }) => required);
+        const productAttributes = super.getConfigurableAttributes();
+        const [
+            productAttributesKeys
+        ] = [Object.keys(productAttributes), Object.values(productAttributes)];
 
-        const configureDownloadableLinks = PRODUCT_TYPE.downloadable && links_purchased_separately === 1;
+        const selectedAttributes = this.getConfigurableAttributes();
+        const [
+            selectedAttributesKeys,
+            selectedAttributesValues
+        ] = [Object.keys(selectedAttributes), Object.values(selectedAttributes)];
```

Comment:
You create an array with two items and then destructure it and assign it to two different variables. Why?
Please assign them to the variables in a common way.